### PR TITLE
Implement QR-Code Service

### PR DESCRIPTION
**Pull request by Frecherenkel60**
## Describe your changes
Implemented QR-Code Service. The workflow is as following:

1. As soon as a ticket is booked, a email with a booking confirmation will be sent. In the mail there'll be a QR-Code for each booked ticket (the QR-Code is the ticket id, since that is unique)
2. If the ticket is scanned the corresponding seat of the ticket needs to be in SeatState.PAID, otherwise an exception will be thrown

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Documentation added?

### CODE REVIEW

File: src/main/java/com/wwi21sebgroup5/cinema/controller/BookingController.java [hunk 1]
```diff
@@ -22,32 +21,47 @@ public class BookingController {
...
-        try{
+    public ResponseEntity<Object> getBookingByBookingId(@RequestParam UUID id) {
+        try {
             Booking requestedBooking = bookingService.findBookingById(id);
             return new ResponseEntity<>(requestedBooking, HttpStatus.OK);
-        }catch(BookingNotFoundException bnfe){
+        } catch (BookingNotFoundException bnfe) {
             return new ResponseEntity<>(bnfe.getMessage(), HttpStatus.NOT_FOUND);
-        }catch(Exception ex){
+        } catch (Exception ex) {
```
Comment (pbahde) [inline src/main/java/com/wwi21sebgroup5/cinema/controller/BookingController.java:30]: Bro triggert dich dass die leerzeichen fehlen?? ;)

File: src/main/java/com/wwi21sebgroup5/cinema/controller/BookingController.java [hunk 2]
```diff
@@ -22,32 +21,47 @@ public class BookingController {
...
-        }catch(UserDoesNotExistException ex){
+        } catch (UserDoesNotExistException ex) {
             return new ResponseEntity<>(ex.getMessage(), HttpStatus.NOT_FOUND);
         }
 
     }
 
+    @PostMapping(path = "/scan", params = "barcode")
+    public ResponseEntity<Object> scanQrCode(@RequestParam String barcode) {
+        try {
```
Comment (pbahde) [inline src/main/java/com/wwi21sebgroup5/cinema/controller/BookingController.java:52]: wieso barcode? als arameter name

**Review by pbahde — APPROVED** [review]
Wenn die QR Code Erstellung so funktioniert passt das so alles :)

File: src/main/java/com/wwi21sebgroup5/cinema/controller/BookingController.java (hunk 2 above)
Comment (Frecherenkel60) [reply src/main/java/com/wwi21sebgroup5/cinema/controller/BookingController.java:52]: code würde denke ich besser passen, oder?

**Pull request merged**